validator/client/src/main/java/tech/pegasys/teku/validator/client/Graffiti.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.validator.client;
+
+import java.nio.file.Path;
+import java.util.Optional;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.tuweni.bytes.Bytes32;
+import tech.pegasys.teku.validator.client.loader.graffiti.GraffitiLoader;
+import tech.pegasys.teku.validator.client.loader.graffiti.GraffitiLoaderException;
+
+/**
+ * A type to hold the current graffiti value. Graffiti can be loaded from 2 places:
+ *
+ * <ul>
+ *   <li>The config file at initialisation
+ *   <li>The graffiti file which will be regularly checked
+ * </ul>
+ *
+ * In the case that the file cannot be read, the graffiti value will remain unchanged from its
+ * previous value.
+ */
+public class Graffiti {
+  private static final Logger LOG = LogManager.getLogger();
+
+  private Bytes32 graffiti;
+  private final Path graffitiFile;
+
+  public Graffiti(final Bytes32 graffiti, final Path graffitiFile) {
+    this.graffiti = graffiti;
+    this.graffitiFile = graffitiFile;
```

Comment:
Thought I left a comment here, but it seems to have been lost.  We'll have to handle the case where one or both of these values are not provided via the CLI.  I'd suggest making these 2 value `Optional`.  Then we can load from file if that argument exists, otherwise we can return the default value.

Reply:
Agreed. Unless I'm missing something I think that's what the code currently is designed to do.

The logic is currently:
Use the new value in the file if possible else fall back to previous value of graffiti
If no value is set the default value will be blank

So I'm unsure exactly what to change here?

Reply:
The issue is I don't think we're handling `null` values safely.  In general, we tend to prefer explicit `Optional`'s in the codebase when a value may be missing.  For example, if the `graffitiFile` is `null`, I believe we'll currently throw a `NullPointerException`.   If we have `Optional`'s, we can do something like: 

```
  private final Optional<Bytes32> defaultGraffiti;
  private final Optional<Path> graffitiFile;

  public Graffiti(final Optional<Bytes32> defaultGraffiti, final Optional<Path> graffitiFile) {
    this.defaultGraffiti = defaultGraffiti;
    this.graffitiFile = graffitiFile;
  }

  public Optional<Bytes32> getGraffiti() {
    return graffitiFile.flatMap(this::loadGraffitiFromFile).or(() -> defaultGraffiti);
  }

  private Optional<Bytes32> loadGraffitiFromFile(final Path path) {
    try {
      return Optional.ofNullable(GraffitiParser.loadFromFile(path));
    } catch (final GraffitiLoaderException e) {
      LOG.error("Loading graffiti from file " + graffitiFile + " failed.", e);
      return Optional.empty();
    }
  }
```

Note - I have changed the behavior a bit here ^.  Instead of keeping the previous value in `graffiti` we just have a static default value to use if the file reading fails.  I think this ends up being a little more intuitive and a little safer:
* If I'm supplying both a graffitiFile and a graffiti value, it seems more intuitive that the static value will be treated as a default rather than potentially being used only on the first block proposal if file reading fails.  
* If you're running multiple validators, you can set a default value to an empty `Bytes32` to avoid potentially linking 2 different validators.  Say someone has a set up that writes new data to the file every slot so that each validator will broadcast unique graffiti.  If a file read fails, rather than using the previous value, we just use an empty value.  This prevents us from broadcasting the same graffiti for different validators.